Sources/bip32-ed25519-swift/bip32-ed25519-swift.swift

```diff
@@ -20,19 +20,33 @@ import Clibsodium
 import Foundation
 import BigInt
 import Foundation
+import JSONSchema
+import MessagePack
 
 public enum KeyContext: UInt32 {
     case Address = 0
     case Identity = 1
 }
 
-enum Encoding {
+public enum Encoding {
     // case cbor
     case msgpack
     case base64
     case none
 }
 
+class DataValidationException: Error {
+    var message: String
+    init(message: String) {
+        self.message = message
+    }
+}
+
+public struct SignMetadata {
+    var encoding: Encoding
+    var schema: [String: Any] // Assuming JSONSchema is a type you've defined elsewhere
```

Comment:
what does `[String: Any]` mean?

Reply:
It's how in Swift JSON objects are represented.

Reply:
and why can't we define a `JSONSchema` type here?

Reply:
I've added a Schema struct that can be provided with a URL and will attempt to load the schema. Added tests as well to show it will fail for malformed schemas.